File: domain/src/main/java/com/github/anrimian/musicplayer/domain/models/playlist/PlayListItem.java
package com.github.anrimian.musicplayer.domain.models.playlist;

import com.github.anrimian.musicplayer.domain.models.composition.Composition;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

public class PlayListItem {

    private final long itemId;

    @Nullable
    private final Long storageItemId;

    @Nonnull
    private final Composition composition;

    public PlayListItem(long itemId,
                        @Nullable Long storageItemId,
                        @Nonnull Composition composition) {
        this.itemId = itemId;
        this.storageItemId = storageItemId;
        this.composition = composition;
    }

    @Nullable
    public Long getStorageItemId() {
        return storageItemId;
    }

    public long getItemId() {
        return itemId;
    }

    @Nonnull
    public Composition getComposition() {
        return composition;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        PlayListItem that = (PlayListItem) o;

        return itemId == that.itemId;
    }

    @Override
    public int hashCode() {
        return (int) (itemId ^ (itemId >>> 32));
    }

    @Override
    public String toString() {
        return "PlayListItem{" +
                "itemId=" + itemId +
                ", composition=" + composition +
                '}';
    }
}

File: domain/src/main/java/com/github/anrimian/musicplayer/domain/models/playlist/PlayListItem.kt
package com.github.anrimian.musicplayer.domain.models.playlist

import com.github.anrimian.musicplayer.domain.models.composition.Composition

class PlayListItem(
    val itemId: Long,
    val composition: Composition
) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as PlayListItem

        if (itemId != other.itemId) return false

        return true
    }

    override fun hashCode(): Int {
        return itemId.hashCode()
    }

    override fun toString(): String {
        return "PlayListItem(itemId=$itemId, composition=$composition)"
    }

}